console/console-init/ui/src/modules/iot-device/components/MangeColumnModal/ManageColumnModal.tsx

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+
+import {
+  Modal,
+  TextContent,
+  TextVariants,
+  Text,
+  Button,
+  DataList,
+  DataListItem,
+  DataListItemRow,
+  DataListCheck,
+  DataListItemCells,
+  DataListCell
+} from "@patternfly/react-core";
+import { getInitialCheckedColumns } from "modules/iot-device/utils";
+
+export interface ICheckedColumns {
+  isDeviceIdChecked: boolean;
+  isConnectionTypeChecked: boolean;
+  isStatusChecked: boolean;
+  isLastUpdatedChecked: boolean;
+  isLastSeenChecked: boolean;
+  isAddedDateChecked: boolean;
+  isMemberOfChecked: boolean;
+  isViaGatewaysChecked: boolean;
+}
+
+interface IManageColumnModalProps {
+  isModalOpen: boolean;
+  handleModalToggle: () => void;
+  setSelectedColumns: (columns: string[]) => void;
+}
+
+const ManageColumnModal: React.FunctionComponent<IManageColumnModalProps> = ({
+  isModalOpen,
+  handleModalToggle,
+  setSelectedColumns
+}) => {
+  const [checkedColumns, setCheckedColumns] = useState<ICheckedColumns>(
+    getInitialCheckedColumns()
+  );
+
+  const handleChange = (checked: boolean, event: any) => {
+    switch (event.target.name) {
```

Comment:
I would like to suggest use switch cases  in small or upper case. Some times It can cause bug. 
switch (event.target.name?.toLowerCase()) {
  case "deviceid":
}